### ISSUE TITLE
Events are uploaded twice using timesketch_import_client with csv or jsonl files.

### ISSUE THREAD

**Issue by S-Nicholas** [google/timesketch#2334]
**Describe the bug**
Events are uploaded twice using timesketch_import_client with csv or jsonl files.

**To Reproduce**
1. Generate csv or json file
2. Use timesketch_importer.py to import the file
3. Events are imported twice in timesketch

**Expected behavior**
Events should be imported only once.

**Additional context**
The issue is located in the ImportStreamer class.
Private property _data_lines is not empty after calling the flush method.
When the ImportStreamer object is destroyed, flush method is called again.
Fix: ImportStreamer._upload_data_buffer method should call _reset method in case of successfull data upload.

**Comment by jaegeral** [comment]
This def needs to be reproduced.

Hope we can fix it in Q3.

**Comment by S-Nicholas** [comment]
Same issue as https://github.com/google/timesketch/issues/2796

**Comment by berggren** [comment]
I cannot seem to reproduce this. I'm importing a CSV with 48300 events with the importer client and I get the expected result.

@S-Nicholas Do you have a stable reproduce?

**Comment by jaegeral** [comment]
I am closing this in favour to [#2796](https://github.com/google/timesketch/issues/2796)
